### PR TITLE
Upgrade MUI v4 to v5

### PR DESCRIPTION
**Pull request by mwodahl**
# PR Details
## Description

This PR upgrades the Material UI version from v4 to v5. The migration guide used for these changes can be found at https://mui.com/material-ui/migration/migration-v4/. 

## How Has This Been Tested?

This has been tested locally using docker-compose to ensure that all components render/function as they previously did. 

## Types of changes

- [ X ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] My changes require new environment variables:
  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ ] My changes require updates to the documentation:
  - [ ] I have updated the documentation accordingly.
- [ X ] My changes require updates and/or additions to the unit tests:
  - [ X ] I have modified/added tests to cover my changes.
- [ X ] All existing tests pass.

### CODE REVIEW

File: webapp/src/App.tsx [hunk 1]
```diff
@@ -44,19 +44,21 @@ const App = () => {
...
-        ) : (
-          <Routes>
-            <Route index element={<Navigate to="dashboard" replace />} />
-            <Route path="dashboard/*" element={<Dashboard />} />
-            <Route path="*" element={<NotFound shouldRedirect={true} />} />
-          </Routes>
-        )}
-      </BrowserRouter>
-    </ThemeProvider>
+    <StyledEngineProvider injectFirst>
```
Comment (jacob6838) [inline webapp/src/App.tsx:47]: Why is this required?

Comment (mwodahl) [reply webapp/src/App.tsx:47]: This ensures that MUI styles don't override any custom styles we've defined. Without the injectFirst prop, MUI styles are added at the end of the <head> element so they may override any custom styles we've defined.

Comment (jacob6838) [reply webapp/src/App.tsx:47]: Is that not the point of specifying an additional theme, to override the original MUI theme? Or am I missing something?

Comment (jacob6838) [reply webapp/src/App.tsx:47]: Looks like this is necessary when running MUI themes with css styles... Unforfunate, but oh well. Hopefully we can migrate these CSS styles to MUI in the future

**Review by jacob6838 — APPROVED** [review]
Tested locally and tests pass, looks great!

**Review by drewjj — APPROVED** [review]
Everything seems to run and tests pass. Looks good to me.

**Review by payneBrandon — APPROVED** [review]
lgtm, tests all pass!
